### DISCUSSION TITLE
Rainbowkit with SIWE

### DISCUSSION THREAD

**Discussion by koolamusic** [rainbow-me/rainbowkit#438] [Ideas]
Hi guys.
great work here.

i am referring to this section on the wagmi docs that implements `SIWE` https://wagmi.sh/examples/sign-in-with-ethereum
and wondering how to achieve same with Raibowkit

**Comment by nickbytes** [comment]
hey @koolamusic – thanks for checking out RainbowKit! You should be able to follow relatively the same steps noted here. RainbowKit replaces the connection UI + simplifies the connection logic. Once you are connected, everything else is just regular wagmi and your backend code. 

If you have a WIP repo to share, the community might be able to help you anywhere you are stuck!

**Reply by koolamusic** [reply]
Thanks, I actually saw that, so there are two recipes I am exploring

Recipe 1: 
I destructure ` const { data, isSuccess, status } = useAccount()` from wagmi `useAccount()` hook, I believe it's what RBkit uses in the background. once desctructured, I add it to a `useEffect` as dependency on the page which subsequently calls my `SIWE` hook to complete that phase. 

This is my version one, dirty and simple to get me moving quick

Recipe 2: 
leverage `<ConnectButton.Custom />` a bit more work on this side, but I also noticed that I have access to the function methods which are called onClick and I could bundle my Connect with SIWE together in one function call. 


I will try to share Recipe 1 and see how we could transform it to recipe 2

**Reply by nickbytes** [reply]
Nice! Yea both those approaches sound promising. We're also thinking of expanding the API a bit here: https://github.com/rainbow-me/rainbowkit/pull/389

**Reply by koolamusic** [reply]
> Nice! Yea both those approaches sound promising. We're also thinking of expanding the API a bit here: https://github.com/rainbow-me/rainbowkit/pull/389

Ohh this gives me joy 😊😂

**Comment by Barabazs** [comment]
I'm also trying to use Rainbowkit with SIWE, but I'm stuck on combining connect+signing as described in the [wagmi docs](https://wagmi.sh/examples/sign-in-with-ethereum#combine-connect-wallet-and-sign--verify-message-steps).
I need to pass a connector to connectAsync(), but afaik Rainbowkit does not expose a connector. 

Did anyone succeed in using this with Rainbowkit?

**Comment by dereknelson** [comment]
also using rainbowkit with SIWE, but in react native - was wondering if there are any resources on best practices for implementing rainbowkit in a webview (or if it's even completely possible/worth it?)

**Comment by DanielSinclair** [comment]
This is first party supported with the [Authentication](https://www.rainbowkit.com/docs/authentication) API and our sister adapter package `@rainbow-me/rainbowkit-siwe-next-auth`. Reference our [Next Auth](https://github.com/rainbow-me/rainbowkit/tree/main/examples/with-next-siwe-next-auth) and [Iron Session](https://github.com/rainbow-me/rainbowkit/tree/main/examples/with-next-siwe-iron-session) examples. Check back soon for an even better SIWE experience.
